### PR TITLE
[2.22] Mobile UCR: use bar graphs instead of line graphs

### PR DESCRIPTION
**Pull request by orangejenny**
Depends on https://github.com/dimagi/commcare-odk/pull/293 which will likely go out in 2.22

screenshot_2015-05-15-13-26-21

### CODE REVIEW

**Comment by orangejenny** [comment]
Some good upcoming small tasks could be
- specify colors for different series (add line-color when creating series in models.py)
- add legend (needs mobile support and then to be added to graph config in models.py)
- add option to do stacked bars (needs mobile support and a way to specify in report config)

File: corehq/apps/userreports/fixtures.py [hunk 1]
```diff
@@ -36,28 +36,7 @@ def _report_to_fixture(report):
...
-            row_elem.append(_element('column', _serialize(row[k]), attrib={'id': k}))
-        rows_elem.append(row_elem)
-        for needed_label in xlabels.keys():
-            if needed_label in row:
-                xlabels[needed_label][str(i)] = _serialize(row[needed_label])
-
-    for column, labels in xlabels.items():
-        report_elem.append(_element('xlabels', json.dumps(labels), attrib={'column': column}))
-
-    report_elem.append(rows_elem)
```
Comment (dannyroberts) [inline corehq/apps/userreports/fixtures.py]: It looks like this removes all the data from the fixture. Is that what you meant to do or did you just mean to get rid of xlabels?

Comment (czue) [inline corehq/apps/userreports/fixtures.py]: (seems unlikely we would want to remove the data)

Comment (orangejenny) [inline corehq/apps/userreports/fixtures.py]: Indeed, I did not mean to delete all data. Should be fixed now.

**Comment by dannyroberts** [comment]
@orangejenny Let me know when 2.22 is released! No point in merging this until then, even if it is only experimental

**Comment by orangejenny** [comment]
@dannyroberts What about for internal experimentation?

**Comment by dannyroberts** [comment]
@czue is anyone using our original version of mobile UCR? If not really, we should probably just merge this and let jenny test with it on prod for 2.22. @orangejenny regardless, you should make sure this feature is only enabled after 2.22. See https://github.com/dimagi/commcare-hq/blob/fdeba03e08d04b4befd41d438dbed60769eaf318/corehq/apps/app_manager/feature_support.py#L84 for that pattern.

**Comment by dannyroberts** [comment]
Though if this is under a flag, then that can always be done later and doesn't need to block this PR...

**Comment by czue** [comment]
i don't think anybody is using it except for demos.

**Comment by czue** [comment]
(probably fine to merge from this perspective - especially if only charts break)

**Comment by orangejenny** [comment]
@dannyroberts I don't quite follow - which piece should be behind 2.22? Pre-2.22, we should keep displaying line graphs? I thought all of mobile UCR was experimental and we were safe to say "If you want to use this, you need to be on the latest version of CommCare" until we get to the point where it's more widely released.

**Comment by czue** [comment]
@orangejenny I think that's fine as long as the new config doesn't crash on the current released version of mobile. is that true?

**Comment by dannyroberts** [comment]
@orangejenny yeah it's experimental and we *definitely* don't need to keep support for the line graphs! I was just saying out loud that at some point we need to just add the commcare version 2.22+ restriction so people will know they have to upgrade to get it. I'm merging this since it's all currently behind a flag. Let me know if you want me to show you what I mean and we can offline.

**Pull request merged**